src/dom-manipulation.js

```diff
@@ -8,31 +8,72 @@ import {
 } from "./utils/utils.js";
 
 export function appendFlightInformationToFlightInfoContainer(flight) {
-  const flightInfo = document.getElementById("flights-info");
-  if (flightInfo) {
-    flightInfo.innerHTML += `
-          <div class="single-flight">
-              <span>${flight[1] || "None"}</span>
-              <span class="media full-screen">${convertMeterPerSecondToKilomentersPerHour(
-                flight[11] ?? 0.0
-              )}m/s</span>
-              <span>${flight[9] ?? 0}m/s</span>
-              <span class="media medium-screen-size">${calculateDirection(
-                flight[10]
-              )}</span>
-              <span>${flight[10] ?? 0}°</span>
-              <span class="media large-screen-size">${flight[7] ?? 0.0}m</span>
-              <button id="${
-                flight[1]
-              }" class="track-button">track flight</button> 
-            </div>`;
+  const flightInfoDiv = document.getElementById("flights-info");
+  const flightButtonExistence = document.getElementById(flight[0] + flight[1]);
+  if (flightInfoDiv && !flightButtonExistence) {
+    createNewFlightInfoRow(flight, flightInfoDiv);
+  } else {
+    appendExistingFlightInfoRow(flight);
+  }
+}
+
+function createNewFlightInfoRow(flight, flightInfoDiv) {
+  flightInfoDiv.innerHTML += `
+  <div id="${flight[0]}" class="single-flight">
+      <span>${flight[1] || "None"}</span>
+      <span class="media full-screen">${convertMeterPerSecondToKilomentersPerHour(
+        flight[11] ?? 0.0
+      )}m/s</span>
+      <span>${flight[9] ?? 0}m/s</span>
+      <span class="media medium-screen-size">${calculateDirection(
+        flight[10]
+      )}</span>
+      <span>${flight[10] ?? 0}°</span>
+      <span class="media large-screen-size">${flight[7] ?? 0.0}m</span>
+      <button id="${
+        flight[0] + flight[1]
+      }" class="track-button">track flight</button> 
+    </div>`;
+}
+
+function appendExistingFlightInfoRow(flight) {
+  const exsitingFlightInfoRow = document.getElementById(flight[0]);
+  if (exsitingFlightInfoRow) {
+    exsitingFlightInfoRow.innerHTML = `
+    <span>${flight[1] || "None"}</span>
+    <span class="media full-screen">${convertMeterPerSecondToKilomentersPerHour(
+      flight[11] ?? 0.0
+    )}m/s</span>
+    <span>${flight[9] ?? 0}m/s</span>
+    <span class="media medium-screen-size">${calculateDirection(
+      flight[10]
+    )}</span>
+    <span>${flight[10] ?? 0}°</span>
+    <span class="media large-screen-size">${flight[7] ?? 0.0}m</span>
+    <button id="${
+      flight[0] + flight[1]
```

Comment:
What happens if `flight[1]` is null?

Reply:
added nullish coalescing, it will now add 0 to the flight[0] in the case of null.